nixos/modules/misc/nixpkgs.nix

```diff
@@ -417,6 +434,11 @@ in
           '';
         }
       ];
+
+    nixpkgs.config.allowUnfreePredicate = lib.mkIf (
```

Comment:
(no action required)
If you could add a type to `allowUnfreePredicate`, something like `functionTo boolByOr`, that would also be useful, but maybe performance matters, so this sort of approach is alright and probably a little faster because the it doesn't assemble a large function that performs a "contrived" linear search.

Reply:
@roberth could you point me to some documentation on how to add types to functions? My investigation where allowUnfreePredicate is defined has stopped at [its type definition](https://github.com/NixOS/nixpkgs/blob/33873b2392e066bea63a2fcf0c882fc487553868/nixos/modules/misc/nixpkgs.nix#L35) and [the definition of `nixpkgs.config`](https://github.com/NixOS/nixpkgs/blob/33873b2392e066bea63a2fcf0c882fc487553868/nixos/modules/misc/nixpkgs.nix#L160) where it is defined as just an attribute set.

How would it have to look to replace this attribute set safely by a module with types without breaking existing configs?

Reply:
Also, how would I write tests for that?